bot/cogs/moderation/infractions.py

```diff
@@ -223,10 +223,13 @@ async def apply_mute(self, ctx: Context, user: Member, reason: t.Optional[str],
         self.mod_log.ignore(Event.member_update, user.id)
 
         async def action() -> None:
-            await user.add_roles(self._muted_role, reason=reason)
-
-            log.trace(f"Attempting to kick {user} from voice because they've been muted.")
-            await user.move_to(None, reason=reason)
+            try:
+                await user.add_roles(self._muted_role, reason=reason)
+            except discord.NotFound:
+                log.info(f"User {user} ({user.id}) left from guild. Can't give Muted role.")
+            else:
+                log.trace(f"Attempting to kick {user} from voice because they've been muted.")
+                await user.move_to(None, reason=reason)
```

Comment:
According to the documentation, this doesn't raise `NotFound`. However, surely it would still fail if the member left the guild. You need to figure out what _exactly_ it raises and only catch that. You may need to look at `HTTPException.code` if the status is not enough of an indicator.